torch/distributed/fsdp/_flat_param.py

```diff
@@ -95,6 +95,13 @@
 # Some value to set padding in tensors to for debuggability
 _FLAT_PARAM_PADDING_VALUE = 42
 
+# Environment variables for disabling the all-gather and reduce-scatter
+# communication ops for ablation studies. Note that without these communication
+# ops the training won't converge, and you probably need to disable correctness
+# checks in your model.
+_FSDP_USE_FAKE_ALL_GATHER = "FSDP_USE_FAKE_ALL_GATHER"
```

Comment:
instead of doing this, I am wondering if you should just make a FakeProcessGroup instead? https://github.com/pytorch/pytorch/blob/main/torch/testing/_internal/distributed/fake_pg.py#L25

Reply:
Thanks for the suggestion! I didn't know the existence of the FakeProcessGroup. Let me try it.

Reply:
My bad. I forgot about `FakeProcessGroup`.

Reply:
Some reference test example in case it's easier to play with it https://github.com/pytorch/pytorch/blob/main/test/distributed/test_fake_pg.py#L82

Reply:
I change to use FakeProcessGroup, as suggested by wanchaol.